### PR TITLE
AWS - S3 Improvements

### PR DESCRIPTION
**Pull request by amshamah419**
## Status
In Progress

## Related Issues
fixes: link to the issue

## Description
Bring up to code standards as well as adding new functionality

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [x] Documentation (demisto/etc#17177)
- [x] Code Review

### CODE REVIEW

File: Integrations/AWS-S3/integration-AWS-S3.py [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+        region=args.get('region'),
+        roleArn=args.get('roleArn'),
+        roleSessionName=args.get('roleSessionName'),
+        roleSessionDuration=args.get('roleSessionDuration'),
+    )
+    path = get_file_path(args.get('entryID'))
+
+    with open(path['path'], 'rb') as data:
+        client.upload_fileobj(data, args.get('bucket'), args.get('key'))
+        demisto.results('File {file} was uploaded successfully to {bucket}'.format(
```
Comment (yuvalbenshalom) [inline Integrations/AWS-S3/integration-AWS-S3.py]: please use py3 format style `f'my format string with{my_var}'`

File: Integrations/AWS-S3/integration-AWS-S3.py [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+
+
+def convert_size(size_bytes):
+    if size_bytes == 0:
+        return "0B"
+    size_name = ("B", "KB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB")
+    i = int(math.floor(math.log(size_bytes, 1024)))
+    p = math.pow(1024, i)
+    s = round(size_bytes / p, 2)
+    return "{} {}".format(s, size_name[i])
```
Comment (yuvalbenshalom) [inline Integrations/AWS-S3/integration-AWS-S3.py]: please use py3 format style

File: Integrations/AWS-S3/integration-AWS-S3.py [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+    client = aws_session(
+        region=args.get('region'),
+        roleArn=args.get('roleArn'),
+        roleSessionName=args.get('roleSessionName'),
+        roleSessionDuration=args.get('roleSessionDuration'),
+    )
+
+    response = client.delete_bucket(Bucket=args.get('bucket').lower())
+    if response['ResponseMetadata']['HTTPStatusCode'] == 204:
+        demisto.results("the Bucket {bucket} was Deleted ".format(bucket=args.get('bucket')))
```
Comment (yuvalbenshalom) [inline Integrations/AWS-S3/integration-AWS-S3.py]: please use py3 format style

File: Integrations/AWS-S3/integration-AWS-S3.py [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+        'Bucket': args.get('bucket').lower(),
+        'Policy': args.get('policy')
+    }
+    if args.get('confirmRemoveSelfBucketAccess') is not None:
+        kwargs.update({'ConfirmRemoveSelfBucketAccess': True if args.get(
+            'confirmRemoveSelfBucketAccess') == 'True' else False})
+
+    response = client.put_bucket_policy(**kwargs)
+    if response['ResponseMetadata']['HTTPStatusCode'] == 200:
+        demisto.results('Successfully applied Bucket policy to {bucket} bucket'.format(bucket=args.get('BucketName')))
```
Comment (yuvalbenshalom) [inline Integrations/AWS-S3/integration-AWS-S3.py]: please use py3 format style

File: Integrations/AWS-S3/integration-AWS-S3.py [hunk 5]
```diff
@@ -0,0 +1,368 @@
...
+
+def delete_bucket_policy_command(args):
+    client = aws_session(
+        region=args.get('region'),
+        roleArn=args.get('roleArn'),
+        roleSessionName=args.get('roleSessionName'),
+        roleSessionDuration=args.get('roleSessionDuration'),
+    )
+    client.delete_bucket_policy(Bucket=args.get('bucket').lower())
+    demisto.results('Policy deleted from {}'.format(args.get('bucket')))
```
Comment (yuvalbenshalom) [inline Integrations/AWS-S3/integration-AWS-S3.py]: please use py3 format style

File: Integrations/AWS-S3/integration-AWS-S3.py [hunk 6]
```diff
@@ -0,0 +1,368 @@
...
+        download_file_command(demisto.args())
+
+    elif demisto.command() == 'aws-s3-list-bucket-objects':
+        list_objects_command(demisto.args())
+
+    elif demisto.command() == 'aws-s3-upload-file':
+        upload_file_command(demisto.args())
+
+except Exception as e:
+    return_error('Error has occurred in the AWS S3 Integration: {error}\n {message}'.format(
```
Comment (yuvalbenshalom) [inline Integrations/AWS-S3/integration-AWS-S3.py]: please use py3 format style

File: Integrations/AWS-S3/integration-AWS-S3.py [hunk 7]
```diff
@@ -0,0 +1,368 @@
...
+    elif demisto.command() == 'aws-s3-download-file':
+        download_file_command(demisto.args())
+
+    elif demisto.command() == 'aws-s3-list-bucket-objects':
+        list_objects_command(demisto.args())
+
+    elif demisto.command() == 'aws-s3-upload-file':
+        upload_file_command(demisto.args())
+
+except Exception as e:
```
Comment (yuvalbenshalom) [inline Integrations/AWS-S3/integration-AWS-S3.py]: you should use the same handling for region/bad address error that you did in ec2

File: Integrations/AWS-S3/integration-AWS-S3.py [hunk 8]
```diff
@@ -0,0 +1,368 @@
...
+import math
+import json
+from datetime import datetime, date
+from botocore.config import Config
+import urllib3.util
+
+# Disable insecure warnings
+urllib3.disable_warnings()
+
+"""PARAMETERS"""
```
Comment (yuvalbenshalom) [inline Integrations/AWS-S3/integration-AWS-S3.py]: where Is the proxy handling

File: Integrations/AWS-S3/integration-AWS-S3.yml [hunk 9]
```diff
@@ -497,113 +275,242 @@ script:
...
 tests:
```
Comment (yuvalbenshalom) [inline Integrations/AWS-S3/integration-AWS-S3.yml]: release notes?

**Comment by amshamah419** [comment]
@kirbles19 The YML is here - https://github.com/demisto/content/blob/aws-s3/Integrations/AWS-S3/integration-AWS-S3.yml

**Comment by kirbles19** [comment]
All done.

**Pull request merged**